File: typescript/src/address/addressConfidence.ts
export enum AddressConfidence {
    Unknown = "Unknown",
    VerifiedMatch = "Verified match",
    MultipleMatches = "Multiple matches",
    TooManyMatches = "Too many matches",
    InteractionRequired = "Interaction required",
    PremisesPartial = "Premises partial",
    StreetPartial = "Street partial",
    VerifiedPlace = "Verified place",
    VerifiedStreet = "Verified street",
    IncompleteAddress = "Incomplete address",
    InsufficientSearchTerms = "Insufficient search terms",
    NoMatches = "No matches"
}

export function lookupConfidence(str?: string): AddressConfidence {
    if (str) {
        return Object.keys(AddressConfidence).find(key => AddressConfidence[key as keyof typeof AddressConfidence] === str) as AddressConfidence | AddressConfidence.Unknown;    
    } else {
        return AddressConfidence.Unknown;
    }
}

File: csharp/DVSClient/DVSClient/address/AddressConfidence.cs
using DVSClient.Common;

namespace DVSClient.Address
{
    public enum AddressConfidence
    {
        [EnumStringValue("Unknown")]
        Unknown,
        [EnumStringValue("Verified match")]
        VerifiedMatch,
        [EnumStringValue("Multiple matches")]
        MultipleMatches,
        [EnumStringValue("Too many matches")]
        TooManyMatches,
        [EnumStringValue("Interaction required")]
        InteractionRequired,
        [EnumStringValue("Premises partial")]
        PremisesPartial,
        [EnumStringValue("Street partial")]
        StreetPartial,
        [EnumStringValue("Verified place")]
        VerifiedPlace,
        [EnumStringValue("Verified street")]
        VerifiedStreet,
        [EnumStringValue("Incomplete address")]
        IncompleteAddress,
        [EnumStringValue("Insufficient search terms")]
        InsufficientSearchTerms,
        [EnumStringValue("No matches")]
        NoMatches
    }
}